### PR TITLE
Fix npe when running in limited config in generic-control-plane mode

### PR DESCRIPTION
**Pull request by mjudeikis**
#### What type of PR is this?

Fix NPE when authorization is not set. Edge case only in generic control plane setup when running custom authorization config. 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controlplane/apiserver/options/options.go [hunk 1]
```diff
@@ -229,7 +229,9 @@ func (o *Options) Complete(alternateDNS []string, alternateIPs []net.IP) (Comple
 	}
 
 	// put authorization options in final state
-	completed.Authorization.Complete()
+	if completed.Authorization != nil {
+		completed.Authorization.Complete()
```
Comment (sttts) [inline pkg/controlplane/apiserver/options/options.go:233]: In other, similar cases, we check for nil in the method.

Comment (mjudeikis) [reply pkg/controlplane/apiserver/options/options.go:233]: TBH not sure what you mean. the only thing above can be setting strict struct empty as its nil'able?

Comment (mjudeikis) [reply pkg/controlplane/apiserver/options/options.go:233]: Internet quick search says - nobody is using it :D

Comment (aojea) [reply pkg/controlplane/apiserver/options/options.go:233]: see line 196
```
func (o *Options) Complete(alternateDNS []string, alternateIPs []net.IP) (CompletedOptions, error) {
	if o == nil {
		return CompletedOptions{completedOptions: &completedOptions{}}, nil
	}
```

you should add the logic here https://github.com/kubernetes/kubernetes/blob/1f40e0916ec41848265b510d15a14be308f9dc60/pkg/kubeapiserver/options/authorization.go#L91-L97

**Pull request merged**